### PR TITLE
feat: add omnect health check as of first implementation in omnect-device-service

### PR DESCRIPTION
**Pull request by HarryWaschkeit**
PR https://github.com/omnect/omnect-device-service/pull/82 introduced first implementation of health check, providing among others means to check systemd services.

Apart from the standard checks in omnect-device-service package this change enables checks for services:
- aziot-certd
- aziot-edged
- aziot-indentityd
- aziot-keyd
- aziot-tpmd
- deviceupdate-agent

### CODE REVIEW

File: recipes-omnect/omnect-device-service/omnect-device-service_0.17.1.bb [hunk 1]
```diff
@@ -6,9 +6,9 @@ inherit cargo
 # DEFAULT_PREFERENCE = "-1"
 
 # how to get omnect-device-service could be as easy as but default to a git checkout:
-# SRC_URI += "crate://crates.io/omnect-device-service/0.17.1"
-SRC_URI += "git://github.com/omnect/omnect-device-service.git;protocol=https;nobranch=1;branch=main"
-SRCREV = "b7d3a4184ea45ae4f510da345c5fe0a42f73a113"
+# SRC_URI += "crate://crates.io/omnect-device-service/0.14.16"
+SRC_URI += "git://github.com/HarryWaschkeit/omnect-device-service.git;protocol=https;nobranch=1;branch=hwt-healthcheck"
```
Comment (JoergZeidler) [inline recipes-omnect/omnect-device-service/omnect-device-service_0.17.1.bb:10]: wrong source URI

Comment (JanZachmann) [reply recipes-omnect/omnect-device-service/omnect-device-service_0.17.1.bb:10]: is there a reason to not create the omnect-device-service PR?

File: recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh [hunk 2]
```diff
@@ -437,6 +437,12 @@ while [[ $1 != "" ]]; do
         $ret 0
         ;;
 
+    --*)
+	error "unrecognized option \"$1\""
+        print_help
+        $ret 1
+	;;
+
```
Comment (mlilien) [inline recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh:445]: what does it have to do with health check and what was the reason you introduced this change?

File: recipes-omnect/omnect-health-check/omnect-health-check.bb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+"
+
+RDEPENDS:${PN} = "bash"
+
+do_install() {
+    # tpmfiles.d
+    install -d ${D}${libdir}/tmpfiles.d
+
+    # create tmpfiles.d entry to (re)create dir + permissions
+    install -m 0644 -D ${WORKDIR}/omnect_service_log.tmpfilesd     ${D}${libdir}/tmpfiles.d/omnect_service_log.conf
```
Comment (mlilien) [inline recipes-omnect/omnect-health-check/omnect-health-check.bb:28]: imho the hole recipe should be part of the omnect-device-service recipe and thus
the content of omnect_service_log.tmpfilesd should be part of https://github.com/omnect/meta-omnect/blob/main/recipes-omnect/omnect-device-service/omnect-device-service/omnect-device-service.tmpfilesd or this file should be renamed to omnect-health-check.tmpfilesd

File: recipes-omnect/omnect-health-check/omnect-health-check/omnect_service_log.tmpfilesd [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Note:
+#   the log directory below can only be written by privileged prozesses, so
+#   any scripts wanting to do so need suitable permissions.
+#   as currently only omnect_service_log.sh is writing to this directory and
+#   only in the context of systemd hooks (PreStart/StopPost) it is sufficient
+#   to instruct systemd to let it run privileged.
+d /run/omnect_health_log 0775 root root -
```
Comment (mlilien) [inline recipes-omnect/omnect-health-check/omnect-health-check/omnect_service_log.tmpfilesd:7]: imho could also be part of /run/omnect-device-service or be renamed to omnect-health-check

File: recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh (hunk 2 above)
Comment (HarryWaschkeit) [reply recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh:445]: yeah, right, it has nothing to do with health check, I just stumbled over that you can give whatever options without the script complaining what I found embarrassing.
I can take that out here, but I find that bad behavior nevertheless which should probably targeted independently from this PR.

File: recipes-omnect/omnect-health-check/omnect-health-check.bb (hunk 3 above)
Comment (HarryWaschkeit) [reply recipes-omnect/omnect-health-check/omnect-health-check.bb:28]: if we want to see health check as part of omnect-device-service (on layer context) I can do that.
@JanZachmann what do you think?

**Review by JanZachmann — CHANGES_REQUESTED** [review]

File: recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh (hunk 2 above)
Comment (HarryWaschkeit) [reply recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh]: removed.

File: recipes-omnect/omnect-health-check/omnect-health-check/omnect_service_log.tmpfilesd (hunk 4 above)
Comment (HarryWaschkeit) [reply recipes-omnect/omnect-health-check/omnect-health-check/omnect_service_log.tmpfilesd]: done

File: recipes-omnect/images/omnect-os-image.bb [hunk 5]
```diff
@@ -74,6 +74,7 @@ IMAGE_INSTALL = "\
     procps \
     sudo \
     systemd-analyze \
+    omnect-health-check \
```
[the remark this answers is not in the document]
Comment (HarryWaschkeit) [reply recipes-omnect/images/omnect-os-image.bb]: is already history due to merge into omnect-device-service.

File: recipes-omnect/omnect-device-service/omnect-device-service_0.17.1.bb (hunk 1 above)
Comment (HarryWaschkeit) [reply recipes-omnect/omnect-device-service/omnect-device-service_0.17.1.bb]: corrected.

File: recipes-omnect/omnect-device-service/omnect-device-service.inc [hunk 6]
```diff
@@ -62,6 +62,26 @@ do_install:append() {
...
+    #        probably all of the health check stuff shoul reside in a dedicated
+    #        folder
+    install -m 0755 -D ${S}/healthcheck/healthchecklib.sh                ${D}/${sbindir}/healthchecklib.sh
+    install -m 0755 -D ${S}/healthcheck/omnect_health_check.sh           ${D}/${sbindir}/omnect_health_check.sh
+    install -m 0755 -D ${S}/healthcheck/omnect_health__coredumps.sh      ${D}/${sbindir}/omnect_health__coredumps.sh
+    install -m 0755 -D ${S}/healthcheck/omnect_health__services.sh       ${D}/${sbindir}/omnect_health__services.sh
+    install -m 0755 -D ${S}/healthcheck/omnect_health__system_running.sh ${D}/${sbindir}/omnect_health__system_running.sh
+    install -m 0755 -D ${S}/healthcheck/omnect_health__timesync.sh       ${D}/${sbindir}/omnect_health__timesync.sh
+    install -m 0755 -D ${S}/healthcheck/omnect_service_log.sh            ${D}/${sbindir}/omnect_service_log.sh
+
```
Comment (JanZachmann) [inline recipes-omnect/omnect-device-service/omnect-device-service.inc:80]: rm empty line?

File: recipes-omnect/omnect-device-service/omnect-device-service/omnect-device-service.tmpfilesd [hunk 7]
```diff
@@ -2,3 +2,5 @@
 d /run/omnect-device-service/ssh-tunnel 0700 ssh_tunnel_user ssh_tunnel_user 1h -
 z /mnt/cert/ssh                         0755 root            root            -  -
 z /mnt/cert/ssh/*                       0644 root            root            -  -
+# create healthcheck log directory with desired permissions
+d /run/omnect_health_log 0775 root root -
```
Comment (JanZachmann) [inline recipes-omnect/omnect-device-service/omnect-device-service/omnect-device-service.tmpfilesd:6]: why habe it twice?
here and recipes-omnect/omnect-device-service/omnect-health-check/omnect_service_log.tmpfilesd

File: recipes-omnect/omnect-device-service/omnect-health-check/omnect_service_log.tmpfilesd [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# Note:
+#   the log directory below can only be written by privileged prozesses, so
+#   any scripts wanting to do so need suitable permissions.
+#   as currently only omnect_service_log.sh is writing to this directory and
+#   only in the context of systemd hooks (PreStart/StopPost) it is sufficient
+#   to instruct systemd to let it run privileged.
+d /run/omnect_health_log 0775 root root -
```
Comment (JanZachmann) [inline recipes-omnect/omnect-device-service/omnect-health-check/omnect_service_log.tmpfilesd:7]: why habe it twice?
here and recipes-omnect/omnect-device-service/omnect-device-service/omnect-device-service.tmpfilesd

Comment (HarryWaschkeit) [reply recipes-omnect/omnect-device-service/omnect-health-check/omnect_service_log.tmpfilesd]: this one was just forgotten to be removed after merging entries in omnect-device-service's tmpfiled file.

**Review by JanZachmann — APPROVED** [review]

**Review by JoergZeidler — APPROVED** [review]

File: recipes-omnect/omnect-device-service/omnect-device-service.inc [hunk 9]
```diff
@@ -1,4 +1,4 @@
-FILESEXTRAPATHS:prepend := "${LAYERDIR_omnect}/files:${THISDIR}/${PN}:"
+FILESEXTRAPATHS:prepend := "${LAYERDIR_omnect}/files:${THISDIR}/${PN}:${THISDIR}/omnect-health-check:"
```
Comment (mlilien) [inline recipes-omnect/omnect-device-service/omnect-device-service.inc:1]: obsolete?

Comment (HarryWaschkeit) [reply recipes-omnect/omnect-device-service/omnect-device-service.inc]: removed.

**Review by mlilien — APPROVED** [review]

**Pull request merged**